src/_pytest/python_api.py

```diff
@@ -60,6 +88,13 @@ def __init__(self, expected, rel=None, abs=None, nan_ok: bool = False) -> None:
     def __repr__(self) -> str:
         raise NotImplementedError
 
+    def repr_compare(self, other_side: Any) -> List[str]:
```

Comment:
Let's make these private -- `ApproxBase` instances are public but we don't want `repr_compare` to be construed as public.

```suggestion
    def _repr_compare(self, other_side: Any) -> List[str]:
```

Reply:
But then, `util.py`'s `_compare_eq_any` would call this method, which is private :(

Reply:
That's not a problem; pytest's calling its own internal code is fine, what we don't want is for external users calling (and depending on) this internal API. 👍

Reply:
sgtm, then